beacon-chain/rpc/eth/validator/handlers.go

```diff
@@ -0,0 +1,120 @@
+package validator
+
+import (
+	"bytes"
+	"net/http"
+	"strconv"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/prysmaticlabs/prysm/v4/beacon-chain/rpc/eth/shared"
+	"github.com/prysmaticlabs/prysm/v4/consensus-types/primitives"
+	"github.com/prysmaticlabs/prysm/v4/encoding/bytesutil"
+	"github.com/prysmaticlabs/prysm/v4/network"
+	ethpbalpha "github.com/prysmaticlabs/prysm/v4/proto/prysm/v1alpha1"
+)
+
+// GetAggregateAttestation aggregates all attestations matching the given attestation data root and slot, returning the aggregated result.
+func (s *Server) GetAggregateAttestation(w http.ResponseWriter, r *http.Request) {
+	attDataRoot := r.URL.Query().Get("attestation_data_root")
+	if attDataRoot == "" {
+		errJson := &network.DefaultErrorJson{
+			Message: "Attestation data root is required",
+			Code:    http.StatusBadRequest,
+		}
+		network.WriteError(w, errJson)
+		return
+	}
+	if !bytesutil.IsHex([]byte(attDataRoot)) {
+		errJson := &network.DefaultErrorJson{
+			Message: "Attestation data root is invalid",
+			Code:    http.StatusBadRequest,
+		}
+		network.WriteError(w, errJson)
+		return
+	}
+	rawSlot := r.URL.Query().Get("slot")
```

Comment:
I feel like this will be a useful generic query parameter maybe it can be broken out into its own utility function